### DISCUSSION TITLE
Is there a good way to mock the fetch function?

### DISCUSSION THREAD

**Discussion by justinmchase** [denoland/deno#18809] [Q&A] [answered]
Do you just stub it or is there a simpler way like [`nock`](https://npmjs.com/nock)? Does nock work with deno?

I also found this which looks really promising, if a little old:
https://deno.land/x/mock_fetch@0.3.0

Is there a std library solution for this?

**Comment by ParitoshBh** [comment, marked as answer]
Wouldn't something like,

```js
import { stub } from "@std/testing/mock";

const fetchStub = stub(globalThis, 'fetch', (url, ...rest) => {
    return Promise.resolve(new Response());
});
```

suffice?
